Kernel/Syscalls/mmap.cpp

```diff
@@ -489,9 +490,42 @@ KResultOr<int> Process::sys$munmap(Userspace<void*> addr, size_t size)
         return 0;
     }
 
-    // FIXME: We should also support munmap() across multiple regions. (#175)
+    // Try again while checkin multiple regions at a time
+    // slow: without caching
+    const auto& regions = space().find_regions_intersecting(range_to_unmap);
+    // if there still no regions found error out
+    if (!regions.size())
+        return EINVAL;
 
-    return EINVAL;
+    // check if any of the regions is invalid in somesort, to not accientally
+    // crash with just half a region map left
+    for (auto* region : regions)
+        if (!region->is_mmap())
+            return EPERM;
+
+    Vector<Region*, 2> new_regions;
+
+    for (auto* old_region : regions) {
+        // if it's a full match we can delete the complete old region
+        if (old_region->range().intersect(range_to_unmap).size() == old_region->size()) {
+            VERIFY(space().deallocate_region(*old_region));
```

Comment:
Please don't put effectful expressions inside `VERIFY()`.
While they are always compiled, it makes the code unnecessarily hard to read.

Reply:
Done.